### PR TITLE
OVMM: Minor episode generation updates

### PR DESCRIPTION
**Pull request by yvsriram**
## Motivation and Context

- Use main_synsetkey instead of wsynsetkey
- Filter episodes having same start and goal receptacle categories 

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py [hunk 1]
```diff
@@ -633,8 +639,10 @@ def add_cat_fields_to_episodes(
         episode["candidate_goal_receps"] = get_candidate_receptacles(
             all_rec_goals, goal_rec_cat
         )
-        if enable_add_viewpoints and not add_viewpoints(
-            sim, episode, obj_idx_to_name, debug_viz
+        if (
+            enable_add_viewpoints
+            and not add_viewpoints(sim, episode, obj_idx_to_name, debug_viz)
+            or start_rec_cat == goal_rec_cat
```
Comment (JHurricane96) [inline habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:645]: Minor: we could move this check a bit ahead to save some time.

**Review by JHurricane96 — APPROVED** [review]

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py [hunk 2]
```diff
@@ -633,8 +639,10 @@ def add_cat_fields_to_episodes(
         episode["candidate_goal_receps"] = get_candidate_receptacles(
             all_rec_goals, goal_rec_cat
         )
-        if enable_add_viewpoints and not add_viewpoints(
-            sim, episode, obj_idx_to_name, debug_viz
+        if (
```
Comment (ykarmesh) [inline habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:642]: Currently this check evaluates 3 conditions. Lets say those conditions are represented by a, b and c. Right now you are doing:
(a and not b or c)
Can you add parenthesis to make the ordering of these evaluations clear? I think you are doing this:
((a and not b) or c)

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py (hunk 1 above)
Comment (ASzot) [reply habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:645]: I agree this check should just be moved earlier to clean up this if statement and avoid calling `add_viewpoints`.

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py (hunk 2 above)
Comment (yvsriram) [reply habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py]: Cleaned this up now

**Comment by yvsriram** [comment]
Fixed pre-commit tests locally. For some reason, these are not passing on CircleCI. Seems like CircleCI is not able to install pre-commit in this branch. @ykarmesh @ASzot do you know why this could be happening? For now, I think we can merge this PR

**Pull request merged**